File: frontend/src/lib/supabase.ts
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseAnonKey) {
  throw new Error(
    "Missing Supabase environment variables. Set NEXT_PUBLIC_SUPABASE_URL and " +
      "NEXT_PUBLIC_SUPABASE_ANON_KEY in frontend/.env (see .env.example).",
  );
}

/**
 * Shared Supabase client for browser and client-component code.
 *
 * Only ever holds the publishable (anon) key, so row-level security applies
 * to every request. Server-side work that needs to bypass RLS belongs in the
 * backend, which uses the secret key.
 */
export const supabase = createClient(supabaseUrl, supabaseAnonKey);

File: backend/app/supabase.py
from supabase import Client, create_client

from app.config import settings

if not settings.supabase_url or not settings.supabase_key:
    raise RuntimeError(
        "Missing Supabase configuration. Set SUPABASE_URL and SUPABASE_KEY in "
        "backend/.env (see .env.example)."
    )

#: Shared Supabase client for server-side use.
#:
#: Configured with the secret key, so requests bypass row-level security.
#: Never expose this client or its key to the frontend.
supabase: Client = create_client(settings.supabase_url, settings.supabase_key)
